projects/bms_carrier/src/current_sense.c

```diff
@@ -1,6 +1,82 @@
 #include "current_sense.h"
 
-// TODO(SOFT-172): Actually implement this method
+#include "ads1259_adc.h"
+#include "bms.h"
+#include "exported_enums.h"
+#include "fault_bps.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static Ads1259Storage s_storage;
```

Comment:
Nit: calling this `s_storage` is kind of confusing since we also have a `storage`, might want to call it `s_ads1259_storage` or something